### DISCUSSION TITLE
Navigation bar should have an option to remain sticky

### DISCUSSION THREAD

**Discussion by SonoSooS** [community/community#87021] [Accessibility]
### Select Topic Area

Product Feedback

### Body

Having to scroll all the way up manually to access the menu is just unacceptable (accessibility or not). But with muscle weakness (+ pain) it's especially insulting having to scroll all the way up from line 6300 (or any line number for that matter) to be able to navigate elsewhere, especially while the notifications bar (with the "back to notifications" button) and the pull request bar (with the "review changes" button) is able to do that, so why not main navigation?  
Also pretty sure the old navigation bar (before the rework) did actually stick to the top of the screen, so I don't know why this new one can't.

Even if possible to access the menu while scrolled down, I just can't see how it's possible, as there are no distinct visual cues, so if not that, this is an accessibility problem by itself.

**Comment by ghostinhershell** [comment]
Hi @SonoSooS, 

Thanks so much for reaching out in the Community. I'll file this feedback with the Accessibility team for further review. In the meantime, if you have any additional information to share on this particular feedback post, feel free to ping me and let me know.

Thanks again!

**Comment by Fitzbert-Fitz** [comment]
Uh boy, this is annoying 😉 . Maybe you could also consider showing a mini version of the header/main menu while scolling down. This would keep the visible workspace space big.

**Reply by SonoSooS** [reply]
Actually, now that you mention it, it could be done what mobile browsers have been doing for years: show the entire menu while scrolling up, and hide it when scrolling down.

Personally I hate this approach, and it tends to have an abysmal implementation by almost everyone, but I *know* that it can be implemented correctly (some websites do implement it in a really comfortable and intuitive way), so it's skill issue that almost nobody can get it right.

**Comment by cehfisher** [comment]
Hello @SonoSooS - thank you for providing this feedback! We wanted to follow up on this issue. Have the recent UI changes for the sticky header resolved this issue for you? If yes, could you please close out this issue? If not, what additional changes are needed? Thanks!

**Reply by SonoSooS** [reply]
(Sorry for the late reply, the notification has turned into a ghost notification, so I've only got the notification now.)

On my side it's still not sticky. I have to still manually scroll all the way up to access the menu, with no button or gesture or anything to open the menu.  
(Please don't make it a gesture though, as those work horribly with touchpad scrolling! If anything, please either make it an option to be permanently sticky, or add in a tiny button to toggle it between open and closed (saved as a cookie or something).)

**Reply by cehfisher** [reply]
Hello @SonoSooS - thank you for following up and confirming this is still an issue for you. We will file this with our Accessibility team so that it can be reviewed and will follow-up here with any additional questions or updates. Thanks again!

**Comment by justfairdev** [comment]
Thank you for the detailed feedback. I completely understand the frustration of having to scroll all the way up—especially when dealing with muscle weakness and pain. It’s important for us to ensure that all parts of the interface are both accessible and user-friendly.

We’re aware that features like the notifications bar and the pull request bar have a sticky behavior, and your suggestion to apply a similar approach to the main navigation is well taken. I'll pass along your feedback to our design and accessibility teams to see how we might improve this in future updates. Your input helps us better understand real-world challenges and improve the experience for everyone.

Thanks again for taking the time to share your thoughts.

**Comment by kylie-sre** [comment]
the title element of this post does exactly what the header nav should do. please actually implement this.

**Reply by cehfisher** [reply]
Hello @kylie-sre - this feature request has already been shared internally with the accessibility team. We will follow-up here with any questions or updates. Thanks!

**Comment by alex-cpm** [comment]
It's been a year since this topic appeared.
Millions of developers swipe their screens just to select the menu item related to the current project. You forgot that the menu should not disappear, the concept of the menu has been around for 40 years and it is still convenient. The vast majority of developers use large screens, not mobile devices.
When will you make the menu not disappear?

**Comment by Mingkwan1** [comment]
I heavily agree with you here. As someone with a weak muscles/bones, this is getting ridiculous sometime and resorting to a auto scroll mouse seems to be the only option. I hope this will get fix soon even though it has been discussed for over a year now

**Comment by digiexchris** [comment]
Any updates? It's so irritating working with github actions that scroll way down, and I need to scroll 9 billion times to get back up to the menu. Your competitors are much more usable in this way, they don't hide the menu at all!

**Reply by cehfisher** [reply]
Sorry @digiexchris - I do not see any updates in the internal tracking issue. As this is considered a "feature request" teams have less incentive to make related changes vs a WCAG violation, for example.

**Reply by digiexchris** [reply]
I just noticed there IS a return to top link in this discussion, but it wasn't obvious to me before. It's just a text link way off to the right past the right margin on a wide monitor. Was it just added? If it was there before, it's hard to find compared to the styles on the main menu.

Also, there's a Up arrow with the word Top in the top right of the window where you browse files in a repo, but not on the repo home Code page. Missing any kind of back to top on Settings and Actions.

Given that this is the same navigation action but it differs between pages, and isn't consistently available (doesn't work on the main Code page viewing a readme, only if you're on the Files page, missing entirely from the Actions page, etc...), this actually may be a WCAG violation 

3.2.3 Consistent Navigation

Seems like maybe a modernization effort is mid-progress? Good time for a review on this subject!

**Reply by Amnat-auyana1998** [reply]
> Any updates? It's so irritating working with github actions that scroll way down, and I need to scroll 9 billion times to get back up to the menu. Your competitors are much more usable in this way, they don't hide the menu at all!
@digiexchris 👀
